### PR TITLE
Enable GA tracking on timeout modal

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
To track users being timed out we needed to supply the survey id and form type. The modal open and closing is now being tracked GA side using show and hide triggers. Because this metadata is only available while the session is still active and once the user is signed out this data is then lost, this PR adds that data to the cookies and then that data is read from the cookie into the data layer for GA to pick up. That way this data is always available to be analysed.

- This has been tested with the service design using GA and they are happy that they can get the data they want in GA in order to track this

### How to review 
- Tests pass
- Schemas launch as expected and new metadata is in the HTML head in dataLayer

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by rmccar** [comment]
[Enable GA tracking on modals](https://trello.com/c/qieSbSj9/5424-enable-ga-tracking-on-modals)

**Review by berroar — DISMISSED** [review]

File: app/survey_config/survey_config.py [hunk 1]
```diff
@@ -71,14 +71,7 @@ def get_footer_legal_links(  # pylint: disable=unused-argument, no-self-use
...
-        if self.schema:
-            data_layer.append(
-                {
-                    key: self.schema.json[key]
-                    for key in ["form_type", "survey_id", "title"]
-                    if key in self.schema.json
-                }
-            )
-        return data_layer
+    def get_additional_data_layer_context(  # pylint: disable=no-self-use
```
Comment (petechd) [inline app/survey_config/survey_config.py:74]: Why `self` in here when not used? Why create a method that always returns empty list (also wrong return type)?

File: app/helpers/language_helper.py [hunk 2]
```diff
@@ -35,8 +35,8 @@ def handle_language(metadata: Optional[MetadataProxy] = None) -> None:
                 schema = load_schema_from_metadata(
                     metadata=metadata, language_code=request_language
                 )
-                if schema.json["title"] != cookie_session.get("survey_title"):
-                    cookie_session["survey_title"] = schema.json["title"]
+                if schema.json["title"] != cookie_session.get("title"):
```
Comment (petechd) [inline app/helpers/language_helper.py:38]: Can we not keep the 'survey_title' instead of  changing  it to 'title'? What is the motivation for this change?

File: app/survey_config/survey_config.py (hunk 1 above)
Comment (rmccar) [reply app/survey_config/survey_config.py:74]: No I think this is correct, it will return the empty list because it is only used to return something when used in census but here we need to return nothing

File: app/helpers/language_helper.py (hunk 2 above)
Comment (rmccar) [reply app/helpers/language_helper.py:38]: This is the title in the cookie and has a different name to differentiate from the `survey_title` in the survey config

File: app/survey_config/survey_config.py (hunk 1 above)
Comment (petechd) [reply app/survey_config/survey_config.py:74]: Yeah it is complicated type hint since we overwrite it in census config, we probably need to keep it as it is now.

**Review by petechd — APPROVED** [review]

File: app/survey_config/survey_config.py [hunk 3]
```diff
@@ -71,14 +71,7 @@ def get_footer_legal_links(  # pylint: disable=unused-argument, no-self-use
...
+        self,
+    ) -> list[dict]:
```
Comment (berroar) [inline app/survey_config/survey_config.py:76]: I would just have this as list since only the child class method adds a dict?
```suggestion
    ) -> list:
```

Comment (rmccar) [reply app/survey_config/survey_config.py:76]: It does here but when the census equivalent is used it returns a list of dicts so I thought it would be confusing if they had different return types?

Comment (rmccar) [reply app/survey_config/survey_config.py:76]: I understand your argument though because the actual function does just return an empty list

Comment (berroar) [reply app/survey_config/survey_config.py:76]: Not sure, I don't think it's confusing so long as the type hint for the census config is correct, as anything that uses this parent class would only ever return an empty list. Won't block it on this anyway

Comment (rmccar) [reply app/survey_config/survey_config.py:76]: Ok I'm happy to make the change

**Review by berroar — APPROVED** [review]
